src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -677,7 +677,7 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
                 case BoundKind.ThisReference: // a special kind of parameter
                 case BoundKind.PseudoVariable:
                     // No temporaries are needed. Just generate local = local + value
-                    return originalLHS;
+                    return VisitExpression(originalLHS);
```

Comment:
>VisitExpression(originalLHS) [](http://example.com/codeflow?start=27&length=28)

Should the expression be visited only if it's a parameter?

Reply:
> Should the expression be visited only if it's a parameter?

No we should visit all the nodes. Then I wouldn't have to fix this bug. Now I will have to check where else similar optimization is going to cause problems.

Reply:
Note that this code path is specific to compound operations. I run into a problem with `++` for example. In other contexts we generally visit everything. So, this code was an exception to the rule.